server/services/nomisPushService.js

```diff
@@ -0,0 +1,29 @@
+const logger = require('../../log.js');
+
+module.exports = nomisClientBuilder => {
+
+    async function pushStatus(bookingId, approvalDecision, systemToken) {
+        const nomisClient = nomisClientBuilder(systemToken);
+        const status = (() => {
```

Comment:
don't really understand this.  You creating a function to work out the status and then calling the function immediately?  Why not just have a case statement, why the need to wrap in a dynamic function?

Reply:
Just another way of doing it. Not uncommon to have a self invoking function in JS. Happy to use Switch if preferred

Reply:
I would prefer:
```javascript
        const bob = {
            Yes: 'Approved',
            No: 'Rejected'
        }
        const status = bob[approvalDecision]
```